app/src/main/java/de/danoeh/antennapod/ui/screen/queue/QueueFragment.java

```diff
@@ -516,6 +517,14 @@ private void loadItems(final boolean restoreScrollPosition) {
                 .observeOn(AndroidSchedulers.mainThread())
                 .subscribe(items -> {
                     queue = items;
+                    if (queue.isEmpty() && DBReader.getTotalEpisodeCount(new FeedItemFilter(FeedItemFilter.NEW)) > 0) {
```

Comment:
This still runs on the main thread (everything in subscribe() is). Please extend the lambda above: `fromCallable(DBReader::getQueue)` to add the loading there.